### PR TITLE
Refactor thumbnail handler

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR refactor the thumbnail handler for simplifying posting external thumbnails.

### CODE REVIEW

File: skyportal/handlers/api/thumbnail.py [hunk 1]
```diff
@@ -3,11 +3,46 @@
...
 from PIL import Image, UnidentifiedImageError
+
 from baselayer.app.access import permissions, auth_or_token
 from ..base import BaseHandler
-from ...models import Obj, Thumbnail
+from ...models import Obj, Thumbnail, User
+
+
+def post_thumbnail(data, user_id, session):
+    """Post source to database.
```
Comment (guynir42) [inline skyportal/handlers/api/thumbnail.py:16]:
```suggestion
    """Post thumbnail to database.
```

File: skyportal/handlers/api/thumbnail.py [hunk 2]
```diff
@@ -3,11 +3,46 @@
...
+
+    user = session.scalar(sa.select(User).where(User.id == user_id))
+
+    obj = session.scalars(Obj.select(user).where(Obj.id == data["obj_id"])).first()
+
+    if obj is None:
+        raise AttributeError(f"Invalid obj_id: {data['obj_id']}")
+
+    try:
+        t = create_thumbnail(data['data'], data['ttype'], data["obj_id"], session)
```
Comment (guynir42) [inline skyportal/handlers/api/thumbnail.py:33]: It looks like `create_thumbnail` does something really similar to this function. Any chance we can combine the two into one function and make less spaghetti?

**Review by guynir42 — COMMENTED** [review]
One thing to consider:

File: skyportal/handlers/api/thumbnail.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/thumbnail.py]: Good point, I combined these two.

File: skyportal/handlers/api/thumbnail.py [hunk 3]
```diff
@@ -29,18 +29,46 @@ def post_thumbnail(data, user_id, session):
...
+            obj_id=data["obj_id"],
+            type=data["ttype"],
+            file_uri=file_uri,
+            public_url=f'/static/thumbnails/{data["obj_id"]}_{data["ttype"]}.png',
+        )
+        with open(file_uri, 'wb') as f:
+            f.write(file_bytes)
+
+        session.add(t)
+        session.flush()
```
Comment (guynir42) [inline skyportal/handlers/api/thumbnail.py:65]:
```suggestion
```

**Review by guynir42 — APPROVED** [review]
Much better!

**Pull request merged**